### PR TITLE
feat: disallow user field expansion for metadata

### PR DESCRIPTION
**Pull request by mortenoh**
* Adds in new user property transformer for BaseIdentifiableObject.user (disallows expansion)
* Does not work for collections (i.e. UserGroup.members)

[DHIS2-8750](https://jira.dhis2.org/browse/DHIS2-8750)

### CODE REVIEW

File: dhis-2/dhis-api/src/test/java/org/hisp/dhis/schema/transformer/UserPropertyTransformerTest.java [hunk 1]
```diff
@@ -59,7 +59,8 @@ public void testUserTransform()
         UserPropertyTransformer transformer = new UserPropertyTransformer();
         UserPropertyTransformer.UserDto userDto = (UserPropertyTransformer.UserDto) transformer.transform( user );
 
-        assertEquals( uuid.toString(), userDto.getId() );
+        // assertEquals( uuid.toString(), userDto.getId() );
```
Comment (luciano-fiandesio) [inline dhis-2/dhis-api/src/test/java/org/hisp/dhis/schema/transformer/UserPropertyTransformerTest.java:62]: commented line

File: dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/schema/transformers/UserPropertyTransformerTest.java [hunk 2]
```diff
@@ -118,13 +136,53 @@ public void testJsonSerializer() throws JsonProcessingException
 
         assertEquals( "usernamea", simple.getUser().getUserCredentials().getUsername() );
         assertEquals( uuid, simple.getUser().getUserCredentials().getUuid() );
+
+        assertNotNull( simple.getUsers() );
+        assertEquals( 4, simple.getUsers().size() );
+
+        /*
+        FieldFilterParams params = new FieldFilterParams(
```
Comment (luciano-fiandesio) [inline dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/schema/transformers/UserPropertyTransformerTest.java:144]: commented out code

**Review by luciano-fiandesio — APPROVED** [review]

File: dhis-2/dhis-api/src/test/java/org/hisp/dhis/schema/transformer/UserPropertyTransformerTest.java (hunk 1 above)
Comment (mortenoh) [reply dhis-2/dhis-api/src/test/java/org/hisp/dhis/schema/transformer/UserPropertyTransformerTest.java:62]: Yes, we will enable this in 2.36 when users are only using uuid as identifiers.

File: dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/schema/transformers/UserPropertyTransformerTest.java (hunk 2 above)
Comment (mortenoh) [reply dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/schema/transformers/UserPropertyTransformerTest.java:144]: Still working on this one :) was commented out to get it in before the soft freeze (this is for property transfor for collections)

**Review by larshelge — APPROVED** [review]

**Pull request merged**